packages/auth/src/errors/AuthError.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { AmplifyError } from '@aws-amplify/core/src/Errors';
+import { ErrorParams } from '@aws-amplify/core/src/types/types';
+
+export class AuthError extends AmplifyError {
+	/**
+	 *
+	 * @param message text that describes the main problem.
+	 * @param underlyingError the underlying cause of the error.
+	 * @param recoverySuggestion suggestion to recover from the error.
+	 *
+	 */
+	constructor({
+		message,
+		name,
+		recoverySuggestion,
+		underlyingError,
+	}: ErrorParams) {
+		super({ message, name, recoverySuggestion, underlyingError });
+	}
```

Comment:
Since this is just a naive extension of `AmplifyError` you can probably just do this for now and let the super handle everything else? Maybe even the documentation above wouldn't need to be repeated

```suggestion
	constructor(params: ErrorParams) {
		super(params);
	}
```